esphome/components/ld2420/ld2420.cpp

```diff
@@ -0,0 +1,506 @@
+  this->set_config_mode_(false);
+  if (restart)
+    this->restart_();
+  ESP_LOGCONFIG(TAG, "LD2420 setup complete.");
+}
+
+void LD2420Component::loop() {
+  // If there is a active send command do not process it here, the send command call will handle it.
+  if (!get_cmd_active_()) {
+    if (!available())
+      return;
+    static uint8_t buffer[2048];
+    static uint8_t rx_data;
+    while (available()) {
+      rx_data = read();
+      this->readline_(rx_data, buffer, sizeof(buffer));
+    }
+  }
+}
+
+/*
+Configure commands - little endian
+
+No command can exceed 64 bytes, otherwise they would need be to be split up into multiple sends.
+
+All send command frames will have:
+  Header = FD FC FB FA, Bytes 0 - 3, uint32_t 0xFAFBFCFD
+  Length, bytes 4 - 5, uint16_t 0x0002, must be at least 2 for the command byte if no addon data.
+  Command bytes 6 - 7, uint16_t
+  Footer = 04 03 02 01 - uint32_t 0x01020304, Always last 4 Bytes.
+Receive
+  Error bytes 8-9 uint16_t, 0 = success, all other positive values = error
+
+Enable config mode:
+Send:
+  UART Tx: FD FC FB FA 04 00 FF 00 02 00 04 03 02 01
+  Command = FF 00 - uint16_t 0x00FF
+  Protocol version = 02 00, can be 1 or 2 - uint16_t 0x0002
+Reply:
+  UART Rx: FD FC FB FA 06 00 FF 01 00 00 02 00 04 03 02 01
+
+Disable config mode:
+Send:
+  UART Tx: FD FC FB FA 02 00 FE 00 04 03 02 01
+  Command = FE 00 - uint16_t 0x00FE
+Receive:
+  UART Rx: FD FC FB FA 04 00 FE 01 00 00 04 03 02 01
+
+Configure system parameters:
+UART Tx: FD FC FB FA 08 00 12 00 00 00 64 00 00 00 04 03 02 01  Set system parms
+Command = 12 00 - uint16_t 0x0012
+
+Configure gate sensitivity parameters:
+UART Tx: FD FC FB FA 0E 00 07 00 10 00 60 EA 00 00 20 00 60 EA 00 00 04 03 02 01
+Command = 12 00 - uint16_t 0x0007
+Gate 0 high thresh = 10 00 uint16_t 0x0010, Threshold value = 60 EA 00 00 uint32_t 0x0000EA60
+Gate 0 low thresh = 20 00 uint16_t 0x0020, Threshold value = 60 EA 00 00 uint32_t 0x0000EA60
+
+*/
+
+void LD2420Component::readline_(uint8_t rx_data, uint8_t *buffer, int len) {
+  static int pos = 0;
+
+  if (rx_data >= 0) {
+    if (pos < len - 1) {
+      buffer[pos++] = rx_data;
+      buffer[pos] = 0;
+    } else {
+      pos = 0;
+    }
+    if (pos >= 4) {
+      if (memcmp(&buffer[pos - 4], &CMD_FRAME_FOOTER, sizeof(CMD_FRAME_FOOTER)) == 0) {
+        ESP_LOGD(TAG, "Rx: %s", format_hex_pretty(buffer, pos).c_str());
+        set_cmd_active_(false);  // Set command state to inactive after responce.
+        this->handle_ack_data_(buffer, pos);
+        pos = 0;
+      } else if (buffer[pos - 2] == 0x0D && buffer[pos - 1] == 0x0A) {
+        this->handle_normal_mode_(buffer, pos);
+        pos = 0;
+      } else if ((pos >= 64 && (memcmp(&buffer[pos - 5], &CMD_FRAME_HEADER, sizeof(CMD_FRAME_HEADER)) == 0) &&
+                  buffer[pos - 1] == 0xFA) &&
+                 get_mode_() == 0) {
+        ESP_LOGD(TAG, "Rx: %s", format_hex_pretty(buffer, pos).c_str());
```

Comment:
Let's maybe make this verbose or even very-verbose along with those below -- otherwise it really makes the logs unnecessarily noisy.

```suggestion
        ESP_LOGVV(TAG, "Rx: %s", format_hex_pretty(buffer, pos).c_str());
```

Reply:
I am not keen on this and at the same time not opposed, here is why.
We default to a log level of DEBUG and we should expect higher levels of logging from that setting. The default should not be DEBUG in my view.
There are no documented defines for what is considered debug data or any sublevels of the same.
In this case we are logging what was sent to the LD2420 which is what I would consider as debug data.
This could be removed since it is available via uart debuging but thats essentially the same debug function.
Verbose is a sublevel and we can move it there but I believe there is a fundametal problem with the amount of and type of logging we are doing.
I would not go with VV since it would mix too much log info from other components.

Thoughts?

Reply:
You can add `debug:` to the `uart` component if you want the bytes logged...

Reply:
Yes I noted that option above, then we could just remove the serial output here and report the higher level command log functions. Whats you thought on default log levels? is there a point at which the default could become WARN. This would also ease up on issue of hitting 50ms blocking with slow baud reate logging.

Reply:
> We default to a log level of DEBUG and we should expect higher levels of logging from that setting.

Agree, but...

> The default should not be DEBUG in my view.

Also agree. However...

The pattern I've noticed in several other components is that, where there's logging to log individual bytes/arrays/messages such as you're doing here, the `VV` level is used.

At the default log level (regardless of what that might be) and because (as Jesse mentioned earlier) you can define the log level on a per-component basis, I guess I just do not expect to see arrays of bytes/messages being logged and it feels as though it's spamming the logs too much.

I guess, if it helps, think of as a `debug` level for users, not for developers.

Reply:
Ok, that helps, best would be to remove the serial byte output lines from this component and provide it via the uart component. It results in the same functionality without more code here.

Reply:
That works -- totally valid solution, too. Thanks for working on this! 🍻